cmd/kube-apiserver/app/server.go

```diff
@@ -395,7 +395,7 @@ func CreateKubeAPIServerConfig(
 
 		// Use the config.GenericConfig.EgressSelector lookup as the transport used by the "proxy" subresources.
 		networkContext := egressselector.Cluster.AsNetworkContext()
-		dialer, err := config.GenericConfig.EgressSelector.Lookup(networkContext)
+		dialer, err := config.GenericConfig.EgressSelector.Lookup(networkContext, "proxy")
```

Comment:
We *could* make these separate constants that are looked up in the egress selector file (similar to Master and Cluster). Not sure if that will improve the clarity, let me know your thoughts.

Reply:
+1 I would prefer enums. We can get better type safety that way.